File: 0501-find-mode-in-binary-search-tree/0501-find-mode-in-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    unordered_map<int, int> mp;
    
    void inorder(TreeNode* root) {
        if (root->left) inorder(root->left);
        mp[root->val]++;
        if (root->right) inorder(root->right);
    }

    vector<int> findMode(TreeNode* root) {
        inorder(root);
        int mx = 0;
        for (auto& x : mp) mx = max(mx, x.second);
        vector<int> ans;
        for (auto& x : mp) {
            if (x.second == mx) ans.push_back(x.first);
        }
        return ans;
    }
};

File: 0501-find-mode-in-binary-search-tree/0501-find-mode-in-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findMode(self, root: Optional[TreeNode]) -> List[int]:
        
        def traverse(node):
            if node != None:
                li.append(node.val)
                traverse(node.left)
                traverse(node.right)
        
        li = []
        traverse(root)
        
        ans = Counter(li)
        max_value = max(ans.values())
        return [key for key, value in ans.items() if value == max_value]